java/arcs/core/data/expression/Evaluator.kt

```diff
@@ -164,42 +168,69 @@ sealed class GlobalFunction(val name: String) {
       toSequence<Any>(args[0]).asIterable().union(
         toSequence<Any>(args[1]).asIterable()
       ).asSequence()
+
+    override fun inferredType(evaluator: TypeEvaluator, args: List<InferredType>) =
+      InferredType.UnionType(args.toSet())
   }
 
   /** Find the maximum of a [Sequence]. */
   object Max : GlobalFunction("max") {
     override fun invoke(evaluator: ExpressionEvaluator, args: List<Any?>) =
       toSequence<Comparable<Comparable<*>>>(args[0]).max()
+
+    override fun inferredType(evaluator: TypeEvaluator, args: List<InferredType>) =
+      widen(*args.toTypedArray())
   }
 
   /** Find the minimum of a [Sequence]. */
   object Min : GlobalFunction("min") {
     override fun invoke(evaluator: ExpressionEvaluator, args: List<Any?>) =
       toSequence<Comparable<Comparable<*>>>(args[0]).min()
+
+    override fun inferredType(evaluator: TypeEvaluator, args: List<InferredType>) =
+      widen(*args.toTypedArray())
   }
 
   /** Find the average of a [Sequence]. */
   object Average : GlobalFunction("average") {
     override fun invoke(evaluator: ExpressionEvaluator, args: List<Any?>) =
       toSequence<Int>(args[0]).average()
+
+    override fun inferredType(evaluator: TypeEvaluator, args: List<InferredType>) =
+      InferredType.Primitive.DoubleType
```

Comment:
The return type of Sequence<Int>.average() is Double

I thought we were gonna loosen these aggregate functions to deal with Sequence<Number> instead of forcing int.